### PR TITLE
feat(QClipboardProxy): add support for images, html and urls

### PR DESCRIPTION
**Pull request by 0x-r4bbit**
- also add discovery properties (hasText/hasImage/hasHtml/hasUrls)
- support for image format is also provided as base64 encoded byte array wrapper

Related: https://github.com/status-im/status-desktop/issues/3395

### CODE REVIEW

**Review by saledjenic — APPROVED** [review]

File: lib/src/Status/QClipboardProxy.cpp [hunk 1]
```diff
@@ -1,14 +1,69 @@
...
+
+QByteArray QClipboardProxy::imageBase64() const
+{
+    if (!hasImage())
+        return {};
+
+    const auto img = image();
+    QByteArray byteArray;
+    QBuffer buffer(&byteArray);
+    img.save(&buffer, "PNG"); // writes the image in PNG format inside the buffer
```
Comment (PascalPrecht) [inline lib/src/Status/QClipboardProxy.cpp:57]: Can we determine whether we really have a PNG? This could be JPG or WEBP as well

**Review by osmaczko — APPROVED** [review]

File: lib/src/Status/QClipboardProxy.cpp (hunk 1 above)
Comment (caybro) [reply lib/src/Status/QClipboardProxy.cpp:57]: Here we always save as PNG but we could yeah, TODO :)

**Comment by PascalPrecht** [comment]
@caybro did the heavy-lifting here. Thank you!

File: lib/src/Status/QClipboardProxy.cpp (hunk 1 above)
Comment (caybro) [reply lib/src/Status/QClipboardProxy.cpp:57]: Might be problematic for GIFs... we would effectively kill the animation :/

Comment (PascalPrecht) [reply lib/src/Status/QClipboardProxy.cpp:57]: FWIW I don't think we support sending GIFs as image messages atm

Comment (PascalPrecht) [reply lib/src/Status/QClipboardProxy.cpp:57]: I stand corrected. It looks like we do

**Pull request merged**